src/EventListener/Doctrine/EntityListener.php

```diff
@@ -27,26 +30,47 @@ public function postUpdate(LifecycleEventArgs $eventArgs): void
         $this->dispatch($eventArgs, static fn (IndexableInterface $entity) => IndexEntity::new($entity));
     }
 
+    public function preRemove(LifecycleEventArgs $eventArgs): void
+    {
+        $indexable = $this->extractIndexableFromEvent($eventArgs);
+
+        if ($indexable instanceof IndexableInterface) {
+            $this->removeIndexableStorage->attach($indexable, $indexable->getId());
+        }
+    }
+
     public function postRemove(LifecycleEventArgs $eventArgs): void
     {
-        $this->dispatch($eventArgs, static fn (IndexableInterface $entity) => RemoveEntity::new($entity));
+        $this->dispatch(
+            $eventArgs,
+            fn (IndexableInterface $entity) => new RemoveEntity($entity::class, $this->removeIndexableStorage[$entity]),
```

Comment:
I guess we need to check if this `$this->removeIndexableStorage[$entity]` exists first?